### PR TITLE
Christmas Crackers Event

### PR DESCRIPTION
**Pull request by TheStig294**

### CODE REVIEW

File: lua/randomat2/events/crackers.lua [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    end
+end
+
+function EVENT:End()
+    net.Start("RandomatCrackersEnd")
+    net.Broadcast()
+
+    -- Remove the crackers and candy cane texture from every weapon in time with the music
+    timer.Simple(5.34, function()
+        for _, ent in ipairs(ents.GetAll()) do
```
Comment (Malivil) [inline lua/randomat2/events/crackers.lua]:
```suggestion
        for _, ent in ents.Iterator() do
```

File: lua/randomat2/events/crackers.lua [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+function EVENT:Begin()
+    -- Gives everyone a christmas cracker, containing a joke, paper hat and random item
+    for _, ply in player.Iterator() do
+        ply:Give(crackerClass)
+        ply:SelectWeapon(crackerClass)
+    end
+
+    -- Apply a candy cane texture to every weapon
+    for _, ent in ipairs(ents.GetAll()) do
```
Comment (Malivil) [inline lua/randomat2/events/crackers.lua]:
```suggestion
    for _, ent in ents.Iterator() do
```

File: gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+    -- Returns an entity the owner is looking at, with a tolerance of a bounding box to search for a player in
+    function SWEP:GetTraceEntity()
+        local owner = self:GetOwner()
+        if not IsValid(owner) then return end
+        local startPos = owner:GetShootPos()
+        local endPos = startPos + (owner:GetAimVector() * self.Range)
+        local lowerBoxBound = Vector(-1, -1, -1) * self.PartnerSearchHitBoxSize
+        local upperBoxBound = Vector(1, 1, 1) * self.PartnerSearchHitBoxSize
+
+        local TraceResult = util.TraceHull({
```
Comment (Malivil) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua]:
```suggestion
        local traceResult = util.TraceHull({
```

File: gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+            start = startPos,
+            endpos = endPos,
+            filter = owner,
+            mask = MASK_SHOT_HULL,
+            mins = lowerBoxBound,
+            maxs = upperBoxBound
+        })
+
+        return TraceResult.Entity
```
Comment (Malivil) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua]:
```suggestion
        return traceResult.Entity
```

File: gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+                partnerAim = partnerAim:Angle()
+                partner:SetEyeAngles(partnerAim)
+                -- Message both players what they have to do
+                local message = "Opening a cracker! Hold left-click and walk backwards!"
+                owner:PrintMessage(HUD_PRINTCENTER, message)
+                partner:PrintMessage(HUD_PRINTCENTER, message)
+            end
+
+            -- Make it so either player can hold left-click and walk backwards to open the cracker, to avoid trolling from the cracker-using player
+            if owner:KeyDown(IN_BACK) or (partner:KeyDown(IN_ATTACK) and partner:KeyDown(IN_BACK)) then
```
Comment (Malivil) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua:349]: Why does only the partner need to hold left-click?

File: gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+
+        timer.Create("TTTCrackerDrawAnimation", 0.01, 25, function()
+            self.ViewmodelDrawAnimModifier = self.ViewmodelDrawAnimModifier + 0.04
+        end)
+
+        return true
+    end
+
+    -- First-person viewmodel
+    function SWEP:GetViewModelPosition(EyePos, EyeAng)
```
Comment (Malivil) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua]: Nit: All local variables should be camelCase, not PascalCase

**Review by Malivil — CHANGES_REQUESTED** [review]

File: gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua (hunk 5 above)
Comment (TheStig294) [reply gamemodes/terrortown/entities/weapons/weapon_ttt_cracker.lua:349]: Because the owner holding left-click has already been checked at line 313

**Review by Malivil — APPROVED** [review]
LGTM

**Pull request merged**